File: src/models/Product.ts
import mongoose, { Schema, models, model } from "mongoose";

export interface IProduct {
  name: string;
  description: string;
  category: string;
  price: number;
  rating: number;
  reviewsCount: number;
  images: string[];
  materials: string[];
  sizes: string[];
  isFeatured: boolean;
  inStock: boolean;
}

const ProductSchema = new Schema<IProduct>(
  {
    name: { type: String, required: true, trim: true },
    description: { type: String, default: "" },
    category: { type: String, required: true },
    price: { type: Number, required: true, min: 0 },
    rating: { type: Number, default: 4.5, min: 0, max: 5 },
    reviewsCount: { type: Number, default: 0, min: 0 },
    images: { type: [String], default: [] },
    materials: { type: [String], default: [] },
    sizes: { type: [String], default: ["One Size"] },
    isFeatured: { type: Boolean, default: false },
    inStock: { type: Boolean, default: true },
  },
  { timestamps: true }
);

export default models.Product || model<IProduct>("Product", ProductSchema);

File: mizhara-backend/models/product.go
package models

import "time"

type Product struct {
	ID                string    `json:"id"`
	Name              string    `json:"name"`
	Description       string    `json:"description"`
	Category          string    `json:"category"`
	CategoryID        string    `json:"categoryId,omitempty"`
	CostPrice         float64   `json:"costPrice,omitempty"`
	Price             float64   `json:"price"`
	Rating            float64   `json:"rating"`
	ReviewsCount      int       `json:"reviewsCount"`
	Images            []string  `json:"images"`
	BannerImage       string    `json:"bannerImage,omitempty"`
	BannerImageMobile string    `json:"bannerImageMobile,omitempty"`
	Materials         []string  `json:"materials"`
	Sizes             []string  `json:"sizes"`
	IsFeatured        bool      `json:"isFeatured"`
	IsActive          *bool     `json:"isActive"`
	StockQuantity     int       `json:"stockQuantity"`
	InStock           bool      `json:"inStock"`
	CreatedAt         time.Time `json:"createdAt"`
	UpdatedAt         time.Time `json:"updatedAt"`
}
